### PR TITLE
Add custom name transformations for appsync_func and glue_

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #793
Fixes: #781

### CODE REVIEW

File: resources.go [hunk 1]
```diff
@@ -2455,3 +2497,13 @@ func updateLegacyModuleNames(prov *tfbridge.ProviderInfo) {
 	updateModuleName("aws_ses_configuration_set", "ConfgurationSet", "ConfigurationSet",
 		sesMod, sesMod, nil)
 }
+
+func TransformWithRandomString(n int) string {
```
Comment (jen20) [inline resources.go]: Does this need to be exported? Or if it is used in multiple places can it go in the bridge instead?

File: resources.go [hunk 2]
```diff
@@ -1090,8 +1105,35 @@ func Provider() tfbridge.ProviderInfo {
...
-			"aws_glue_catalog_table":          {Tok: awsResource(glueMod, "CatalogTable")},
+			"aws_glue_catalog_database": {
+				Tok: awsResource(glueMod, "CatalogDatabase"),
+				Fields: map[string]*tfbridge.SchemaInfo{
+					"name": {
+						Default: &tfbridge.DefaultInfo{
+							From: tfbridge.FromName(tfbridge.AutoNameOptions{
+								Transform: func(name string) string {
+									newName := fmt.Sprintf("%s_%s", name, TransformWithRandomString(8))
+									return strings.ToLower(newName)
```
Comment (lukehoban) [inline resources.go]: Why do we need this Transform?  Can't we parameterize the AutoNamer to do this without custom code here?  (OR is there a feature we should add there to handle this)?

**Comment by EvanBoyle** [comment]
Looks good to me! Do we have automated tests for this sort of thing? Would be nice to execute that logic and vet it against `^[a-z]+(_[a-z]+)*$`

File: resources.go [hunk 3]
```diff
@@ -250,7 +252,20 @@ func Provider() tfbridge.ProviderInfo {
...
 				},
 			},
-			"aws_appsync_function":    {Tok: awsResource(appsyncMod, "Function")},
+			"aws_appsync_function": {
+				Tok: awsResource(appsyncMod, "Function"),
+				Fields: map[string]*tfbridge.SchemaInfo{
+					"name": {
+						Default: &tfbridge.DefaultInfo{
+							From: tfbridge.FromName(tfbridge.AutoNameOptions{
+								Separator: "-",
```
[the remark this answers is not in the document]
Comment (stack72) [reply resources.go]: D'oh! that should be an `_` - good spot!

File: resources.go (hunk 2 above)
Comment (stack72) [reply resources.go]: This transform stops adding any random numbers to the ID. This isn't something we have needed before. My thoughts are, if this needs moved to be a feature of it's own in the AutoNaming logic, then we will do so

File: resources.go (hunk 1 above)
Comment (stack72) [reply resources.go]: Fixed!

**Pull request merged**